woocommerce/woocommerce-gutenberg-products-block
# Restore back-compat loading of assets registry switch which ensures wcSettings is only loaded when necessary

Fixes #1020 

See #1020  for further details.  In this pull:

- updated `@woocommerce/components` to the latest version (`4.0.0`) which has `wcSettings` decoupled so no more errors when back-compat code isn't loaded.
- restored the back-compat switch in `Automattic\WooCommerce\Domain\Bootstrap`
- fixed a bug in `AssetDataRegistry` and added tests to guard against regressions.

## To Test

Testing of this is fairly involved to ensure that there is appropriate back-compatibility between different versions of woo-blocks and woo-admin loaded.  So for each scenario the following needs verified:

- there are no console errors related to `wcSettings`
- the functionality of blocks and woo-admin is as expected.
- for versions of woo-admin that have onboarding, make sure onboarding works as expected.
- All tests should use this branch of Woo-Blocks.

| Scenario # | WooAdmin Version | Other notes |
| -------- | ------------- | ------------- |
| 1 | Not Active | - also verify `wcSettings` is only loaded on routes needing it. |
| 2 | Master | - `wcSettings` should only be loaded on routes needing it (including wc-admin routes).
| 3 | 0.19.0 | - `wcSettings` will load on every route.
| 4 | 0.20.0 | - `wcSettings` only loaded on routes needing it (including wc-admin routes).

### Changelog

> Don't enqueue wcSettings unless the route requires it.
